codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoder.java

```diff
@@ -408,7 +419,7 @@ public boolean merge(ChannelHandlerContext ctx, Http2RemoteFlowController.FlowCo
         private final short weight;
         private final boolean exclusive;
 
-        public FlowControlledHeaders(Http2Stream stream, Http2Headers headers, int streamDependency, short weight,
+        FlowControlledHeaders(Http2Stream stream, Http2Headers headers, int streamDependency, short weight,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Constructor has 8 parameters, which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS00107)